pkg/controller/builder.go

```diff
@@ -102,10 +106,28 @@ func (b *Builder) BuildControllers(subDirectories ...string) (map[string]factory
 				return nil
 			}
 
+			if manifestGVK.Kind == "ClusterCatalog" && manifestGVK.Group == catalogdv1alpha1.GroupVersion.Group {
```

Comment:
IMO we should add code comment around why creating controllers for each catalog is not an issue.